packages/angular/lib/index.ts

```diff
@@ -0,0 +1,82 @@
+import { APP_INITIALIZER } from '@angular/core'
+import { NavigationCancel, NavigationEnd, NavigationError, NavigationStart, Router } from '@angular/router'
+import {
+  onSettle,
+  type RouteChangeSpan,
+  type RoutingProvider,
+  type StartRouteChangeCallback
+} from '@bugsnag/browser-performance'
+
+let globalRouterRef: Router | undefined
+
+export const bugsnagBootstrapper = {
+  provide: APP_INITIALIZER,
+  useFactory: (router: Router) => {
+    globalRouterRef = router
+
+    return () => {}
+  },
+  multi: true,
+  deps: [Router]
+}
+
+export class AngularRoutingProvider implements RoutingProvider {
+  resolveRoute (url: URL): string {
+    if (globalRouterRef) {
+      let route = globalRouterRef.routerState.snapshot.root
+      let path = ''
+
+      while (route.firstChild) {
+        route = route.firstChild
+
+        if (route.routeConfig?.matcher) {
+          path += '/<custom URL matcher>'
+        } else {
+          path += '/' + route.routeConfig?.path
+        }
+      }
+
+      if (path) {
+        return path
+      }
+    }
+
+    const base = document.baseURI.replace(window.origin, '')
+    return url.pathname.replace(base, '') || '/'
+  }
+
+  listenForRouteChanges (startRouteChangeSpan: StartRouteChangeCallback): void {
+    // wait for the router to be available
+    if (!globalRouterRef) {
+      setTimeout(() => { this.listenForRouteChanges(startRouteChangeSpan) }, 50)
+      return
+    }
```

Comment:
it would remove the need for this function to recursively call itself, but I can imagine it's probably not a huge saving. I may have worded it badly of course!